x-pack/plugins/enterprise_search/public/applications/shared/telemetry/send_telemetry.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect } from 'react';
+
+import { HttpHandler } from 'src/core/public';
+import { KibanaContext, IKibanaContext } from '../../index';
+
+interface ISendTelemetryProps {
+  action: 'viewed' | 'error' | 'clicked';
+  metric: string; // e.g., 'setup_guide'
+}
+
+interface ISendTelemetry extends ISendTelemetryProps {
+  http(): HttpHandler;
+  product: 'app_search' | 'workplace_search' | 'enterprise_search';
+}
+
+/**
+ * Base function - useful for non-component actions, e.g. clicks
+ */
+
+export const sendTelemetry = ({ http, product, action, metric }: ISendTelemetry) => {
+  return http
+    .put(`/api/${product}/telemetry`, {
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({ action, metric }),
+    })
+    .catch(error => {
+      throw new Error('Unable to send telemetry');
```

Comment:
What happens to these errors upstream -- does the user notice anything wrong?

Reply:
These should just show up as console errors, but shouldn't impact core behavior (I think that's 'working as intended', telemetry generally shouldn't cause breaking behavior or the app to break as it's a nice to have vs core UX kind of thing).